src/vertex_cover.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    vertex_cover!(
+        model, g;
+        var_name
+    )
+
+Modify a JuMP model by adding the variable, constraints and objective necessary to compute a minimum vertex cove of an undirected graph.
+
+The vertex cover indicator variable will be named `var_name`
+"""
+function min_vertex_cover!(model::Model, g::AbstractGraph; var_name)
+    if is_directed(g)
+        throw(ArgumentError("The graph must not be directed"))
+    end
+
+    ver = collect(vertices(g))
+    edge_tuples = [(src(ed), dst(ed)) for ed in edges(g)]
+
+    f = @variable(model, [ver]; integer=true, base_name=String(var_name))
+    model[Symbol(var_name)] = f
+
+    for v in ver
+        @constraint(model, f[v] >= 0)
+        @constraint(model, f[v] <= 1)
+    end
+
+    for (u, v) in edge_tuples
```

Comment:
You don't need to pre-allocate the tuples, you can just iterate over the edges here (microscopic performance win)